packages/runtime/runtime-definitions/src/dataStoreContext.ts

```diff
@@ -180,6 +180,12 @@ export interface IFluidDataStoreChannel extends
      */
     processSignal(message: any, local: boolean): void;
 
+    /**
+     * Generates a snapshot of the given data store
+     * @deprecated in 0.22 summarizerNode
```

Comment:
if it was deprecated so long ago, why do we need it?

Reply:
Also, I prefer to remove things from interfaces right away, but keep implementation. That forces consumers to move away (though that assumes that move is not expensive, i.e. can be done as part of consuming new version)

Reply:
It was deprecated but we were still using it internally across the container runtime / data store runtime boundary. Container runtime's snapshot was calling data store context's snapshot which was calling this.
My change removes data store context's snapshot and the call to it. So now, its actually deprecated internally as well.

Reply:
You are right. It can be removed from the interface. Removed it.